dotCMS/src/main/java/com/dotmarketing/util/ExportStarterUtil.java

```diff
@@ -0,0 +1,305 @@
+import com.dotmarketing.portlets.workflows.util.WorkflowImportExportUtil;
+import com.dotmarketing.tag.model.Tag;
+import com.dotmarketing.tag.model.TagInode;
+import com.google.common.collect.ImmutableList;
+import com.liferay.portal.ejb.ImageLocalManagerUtil;
+import com.liferay.portal.model.Company;
+import com.liferay.portal.model.User;
+import com.liferay.util.FileUtil;
+
+public class ExportStarterUtil {
+
+    
+
+    final File outputDirectory;
+    
+    public ExportStarterUtil() {
+        
+        outputDirectory = new File(ConfigUtils.getBackupPath() + File.separator + "backup_" + System.currentTimeMillis());
+        outputDirectory.mkdirs();
+       
+        
+    }
+    
+    public File createStarterWithAssets() throws FileNotFoundException, IOException {
+        
+        moveAssetsToBackupDir();
+        return createStarterData() ;
+        
+    }
+    
+    public File createStarterData() {
+        createXMLFiles() ;
+        
+        return outputDirectory;
+        
+        
+    }
+    
+    
+    /**
+     * This method will pull a list of all tables /classed being managed by
+     * hibernate and export them, one class per file to the backupTempFilePath
+     * as valid XML. It uses XStream to write the xml out to the files.
+     *
+     * @throws ServletException
+     * @throws IOException
+     * @author Will
+     * @throws DotDataException
+     */
+    @CloseDBIfOpened
+    private void createXMLFiles()  {
+
+        Logger.info(this, "Starting createXMLFiles into " + outputDirectory);
+
+        Set<Class> _tablesToDump = new HashSet<Class>();
+        try {
+
+            /* get a list of all our tables */
+            //Including Identifier.class and Language.class because it is not mapped with Hibernate anymore
+            _tablesToDump.add(Identifier.class);
+            _tablesToDump.add(Language.class);
+            _tablesToDump.add(Relationship.class);
+            _tablesToDump.addAll(HibernateUtil.getSession().getSessionFactory().getAllClassMetadata().keySet());
+
+            _tablesToDump.removeIf(c->c.equals(Inode.class));
+            _tablesToDump.removeIf(c->c.equals(Clickstream.class));
+            _tablesToDump.removeIf(c->c.equals(ClickstreamRequest.class));
+            _tablesToDump.removeIf(c->c.equals(Clickstream404.class));
+            _tablesToDump.removeIf(c->c.equals(Structure.class));
+            _tablesToDump.removeIf(c->c.equals(Field.class));
+            _tablesToDump.removeIf(c->c.equals(WorkflowHistory.class));
+            _tablesToDump.removeIf(c->c.equals(PermissionReference.class));
+            
+            _tablesToDump.removeIf(t->t.getName().startsWith("Dashboard"));
+            _tablesToDump.removeIf(t->t.getName().contains("HBM"));
+
+
+            HibernateUtil _dh = null;
+            DotConnect dc = null;
+            List _list = null;
+            File _writing = null;
+
+
+            for (Class clazz : _tablesToDump) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'clazz' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)